agent/connect/ca/provider_consul.go

```diff
@@ -617,19 +617,13 @@ func (c *ConsulProvider) incrementAndGetNextSerialNumber() (uint64, error) {
 
 // generateCA makes a new root CA using the current private key
 func (c *ConsulProvider) generateCA(privateKey string, sn uint64, rootCertTTL time.Duration) (string, error) {
-	stateStore := c.Delegate.State()
-	_, config, err := stateStore.CAConfig(nil)
-	if err != nil {
-		return "", err
-	}
-
 	privKey, err := connect.ParseSigner(privateKey)
 	if err != nil {
 		return "", fmt.Errorf("error parsing private key %q: %s", privateKey, err)
 	}
 
 	// The URI (SPIFFE compatible) for the cert
-	id := connect.SpiffeIDSigningForCluster(config)
```

Comment:
We already store the `clusterID` on `ConsulProvider` from the call to `Configure`, so we can use that directly instead of fetching the full config from the state store. This removes the only caller to `Delegate.State()` that was using something other than the provider config.

In the future we can reduce the interface here to a call that only returns the provider config.